util/topgen/intermodule.py

```diff
@@ -390,6 +387,54 @@ def check_intermodule_field(obj: OrderedDict, prefix: str = "") -> int:
     return error
 
 
+def find_otherside_modules(topcfg: OrderedDict, m,
+                           s) -> List[Tuple[str, str, str]]:
+    """Find far-end port based on given module and signal name
+    """
+    signame = "{}.{}".format(m, s)
+    for req, rsps in topcfg["inter_module"]["connect"].items():
+        if req.startswith(signame):
+            # return rsps after splitting module instance name and the port
+            result = []
+            for rsp in rsps:
+                rsp_m, rsp_s, rsp_i = filter_index(rsp)
+                result.append(('connect', rsp_m, rsp_s))
+            return result
```

Comment:
when req matches, should we return req which is (hmac, tl), rather than rsp (main, tl_hmac)

```
      hmac.tl:
      [
        main.tl_hmac
      ]
```

Reply:
mem rom and ram_main use special instance names:`u_tl_adapter_rom` and `u_tl_adapter_ram_main`
but they're not indicated in `["inter_module"]["connect"]`
how can we handle them?
```
      rom.tl:
      [
        main.tl_rom
      ]
      ram_main.tl:
      [
        main.tl_ram_main
      ]
```

Reply:
For the first question, The function will return `hmac`, `tl` if arguments are `main` and `tl_hmac`. My assumption of use of this function is to find which IP/port is connected from the port given by the argument. For example, If you only have Xbar ports (in this case `main` xbar and `tl_hmac` port) and want to know which IP the `main.tl_hmac` is connected to, then the function is called as `find_otherside_modules(topcfg, 'main', 'tl_hmac')` then the result will be `[('connect', 'hmac', 'tl)]`.

Reply:
Second one, you are right. as of now, it returns not exact module yet. We will soon have comportable memory IPs too.

Reply:
I see. Thanks for the explanation.